### PR TITLE
Allow search by service name in services list page

### LINKED ISSUE #1511: Search servicesList page using plain text
Currently, the servicesList page does allow filtering by resource attributes but when the number of services become large then it becomes difficult to find a service. 
This is a feature request to enable searching a service by text.

### PR DESCRIPTION
**Pull request by srikanthccv**
Fixes #1511 

This adds an option to table search on service name column

https://www.loom.com/share/a3bb05725cc54faaa6a0a2a0d050a1c2

### CODE REVIEW

**Comment by srikanthccv** [comment]
cc @pranay01 please review

**Comment by palashgdev** [comment]
@srikanthccv can you please have a look over this one https://ant.design/components/table/#components-table-demo-custom-filter-panel

**Comment by pranay01** [comment]
@srikanthccv does it only allow for `search` by service name? What if a user wants to sort by service name

**Comment by srikanthccv** [comment]
@pranay01 Yes, right now it only allows search by service name. The feature request was the ability to search for service. Would you like to see the sort option as well? I thought people would only sort by RED numbers and that's why we don't have the sort operation for names today.

**Comment by ankitnayan** [comment]
@pranay01 @srikanthccv the implementation for the scope of the issue seems fine. By default IMO we sort by `p99` from the backend.

**Comment by srikanthccv** [comment]
@palashgdev Updated, PTAL.

File: frontend/src/container/MetricsTable/index.tsx [hunk 1]
```diff
@@ -1,9 +1,13 @@
-import Table, { ColumnsType } from 'antd/lib/table';
+/* eslint-disable react/no-unstable-nested-components */
```
Comment (palashgdev) [inline frontend/src/container/MetricsTable/index.tsx:1]: @srikanthccv can you please remove this eslint

File: frontend/src/container/MetricsTable/index.tsx [hunk 2]
```diff
@@ -37,21 +44,113 @@ function Metrics(): JSX.Element {
...
+		setSearchText(selectedKeys[0]);
+		setSearchedColumn(dataIndex);
+	};
+
+	const handleReset = (clearFilters): void => {
+		clearFilters();
+		setSearchText('');
+	};
+
+	// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
```
Comment (palashgdev) [inline frontend/src/container/MetricsTable/index.tsx:58]: please remove this one also

File: frontend/src/container/MetricsTable/index.tsx [hunk 3]
```diff
@@ -37,21 +44,113 @@ function Metrics(): JSX.Element {
...
+					ref={searchInput}
+					placeholder={`Search ${dataIndex}`}
+					value={selectedKeys[0]}
+					onChange={(e): void =>
+						setSelectedKeys(e.target.value ? [e.target.value] : [])
+					}
+					onPressEnter={(): void => handleSearch(selectedKeys, confirm, dataIndex)}
+					style={{
+						marginBottom: 8,
+						display: 'block',
```
Comment (palashgdev) [inline frontend/src/container/MetricsTable/index.tsx:81]: why explicitly `block` ?

**Review by palashgdev — CHANGES_REQUESTED** [review]
Small comments

File: frontend/src/container/MetricsTable/index.tsx (hunk 2 above)
Comment (srikanthccv) [reply frontend/src/container/MetricsTable/index.tsx:58]: I might need help on this. This was showing not to return jsx element from the following filter functions. What are the ways to achieve that.

File: frontend/src/container/MetricsTable/index.tsx (hunk 1 above)
Comment (srikanthccv) [reply frontend/src/container/MetricsTable/index.tsx:1]: I might need help on this. This was showing not to return jsx element from the following filter functions. What are the ways to achieve that.

File: frontend/src/container/MetricsTable/index.tsx (hunk 3 above)
Comment (srikanthccv) [reply frontend/src/container/MetricsTable/index.tsx:69]: It was overflowing the table otherwise

**Review by pranshuchittora — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/32242596/186085986-6e5ac300-593e-4403-bdbc-12fac1bc16a3.mov

File: frontend/src/container/MetricsTable/index.tsx [hunk 4]
```diff
@@ -27,6 +30,81 @@ function Metrics(): JSX.Element {
...
+
+	const handleReset = (clearFilters: () => void): void => {
+		clearFilters();
+	};
+
+	const FilterIcon = useCallback(
+		({ filtered }) => (
+			<SearchOutlined
+				style={{
+					color: filtered ? '#1890ff' : undefined,
```
Comment (pranshuchittora) [inline frontend/src/container/MetricsTable/index.tsx:45]: Instead of hardcoding the color, pls use the @antd/colors package

**Review by pranshuchittora — APPROVED** [review]

**Review by palashgdev — APPROVED** [review]

**Pull request merged**